compiler/src/dotty/tools/dotc/util/Sandbox.scala

```diff
@@ -38,15 +38,18 @@ object Sandbox {
 
   object SandboxSecurityManager {
     @volatile private[this] var running: Int = 0
+    private def getRunning = running
+    private def incRunning() = running += 1
+    private def decRunning() = running -= 1
```

Comment:
@nicolasstucki FWIW: if the object is shared and you want `incRunning` to be atomic, you do need `synchronized`, `volatile` is powerless to keep together the read and the write as a single atomic action.

Reply:
I knlow, I stumbled with some other bug and I'm just trying to move things too see what fixes it.

Reply:
BTW, the atomicity was given by the `synchronized` block in the only method that could access `incRunning`.